sui-execution/latest/sui-move-natives/src/object_runtime/mod.rs

```diff
@@ -543,6 +543,16 @@ impl ObjectRuntimeState {
             })
             .collect::<Vec<_>>();
         for id in remaining_by_value_objects {
+            if input_objects
+                .get(&id)
+                .is_some_and(|owner| owner.is_shared())
+            {
+                return Err(ExecutionError::new(
+                    ExecutionErrorKind::SharedObjectOperationNotAllowed,
+                    Some(format!("Wrapping shared object {} not allowed", id).into()),
```

Comment:
After seeing through the changes, it seems that supporting shared object wrapping is in fact trivial and requires no extra change, is my assessment correct? Should we just support it anyway to make the code simpler?

Reply:
I had thought that wrapping/unwrapping was more of an issue on the scheduling/consensus side of things, not Move.

Reply:
I will collect feedback from users and if there is a desire for this feature will simplify here and work through that. It is definitely out of scope for this PR.

Reply:
The problem is that if you can wrap, you can also unwrap, and now there can be an object that has the same id as a previously deleted shared object. This may be okay but it would require some serious thought to be sure it doesn't cause safety issues.